src/test/java/org/folio/logintest/LoginWithExpiryTest.java

```diff
@@ -0,0 +1,320 @@
+
+  private static final String TENANT_DIKU = "diku";
+  private static final String LOGIN_WITH_EXPIRY_PATH = "/authn/login-with-expiry";
+  private static final String CRED_PATH = "/authn/credentials";
+  private static final String UPDATE_PATH = "/authn/update";
+
+  @BeforeClass
+  public static void setup(final TestContext context) throws Exception {
+    vertx = Vertx.vertx();
+
+    int port = NetworkUtils.nextFreePort();
+    int mockPort = NetworkUtils.nextFreePort();
+
+    DeploymentOptions options = new DeploymentOptions().setConfig(
+        new JsonObject()
+            .put("http.port", port));
+
+    DeploymentOptions mockOptions = new DeploymentOptions().setConfig(
+        new JsonObject()
+            .put("port", mockPort));
+
+    PostgresClient.setPostgresTester(new PostgresTesterContainer());
+    PostgresClient.getInstance(vertx);
+
+    spec = new RequestSpecBuilder()
+        .setContentType(ContentType.JSON)
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.URL, "http://localhost:" + mockPort)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.TOKEN, "dummy.token")
+        .addHeader(XOkapiHeaders.REQUEST_TIMESTAMP, String.valueOf(new Date().getTime()))
+        .build();
+
+    specWithoutUrl = new RequestSpecBuilder()
+        .setContentType(ContentType.JSON)
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.TOKEN, "dummy.token")
+        .addHeader(XOkapiHeaders.REQUEST_TIMESTAMP, String.valueOf(new Date().getTime()))
+        .build();
+
+    specWithoutToken = new RequestSpecBuilder()
+        .setContentType(ContentType.JSON)
+        .setBaseUri("http://localhost:" + port)
+        .addHeader(XOkapiHeaders.URL, "http://localhost:" + mockPort)
+        .addHeader(XOkapiHeaders.TENANT, TENANT_DIKU)
+        .addHeader(XOkapiHeaders.REQUEST_TIMESTAMP, String.valueOf(new Date().getTime()))
+        .build();
+
+    TenantAttributes ta = new TenantAttributes().withModuleTo("mod-login-1.1.0");
+    vertx.deployVerticle(Mocks.class.getName(), mockOptions)
+        .compose(res -> vertx.deployVerticle(RestVerticle.class.getName(), options))
+        .compose(res -> TestUtil.postSync(ta, TENANT_DIKU, port, vertx))
+        .onComplete(context.asyncAssertSuccess());
+  }
+
+  @Test
+  public void testLoginWithExpiry(final TestContext context) {
+    RestAssured.given()
+        .spec(spec)
+        .body(credsObject1.encode())
+        .when()
+        .post(CRED_PATH)
+        .then()
+        .log().all()
+        .statusCode(201);
+
+    RestAssured.given()
+        .spec(specWithoutToken)
+        .body(credsNoPassword.encode())
+        .when()
+        .post(LOGIN_WITH_EXPIRY_PATH)
+        .then()
+        .log().all()
+        .statusCode(400)
+        .contentType("text/plain")
+        .body(is("Missing Okapi token header"));
```

Comment:
Why should login require a token? Only X-Okapi-Tenant is needed.

Reply:
I agree that it doesn't make much sense to have this. The reason it is here is that replicates the (possibly wrong) unit test from the `/login` endpoint from [here](https://github.com/folio-org/mod-login/blob/9ca2e3472f7e6f695d37d657979bfe44403a3575/src/test/java/org/folio/logintest/RestVerticleTest.java#L403).

I'll remove the test.